eng/scripts/Language-Settings.ps1

```diff
@@ -348,6 +369,43 @@ function Get-DocsCiLine ($item) {
   return $line
 }
 
+function EnsureCustomSource($package) {
+  # $PackageSourceOverride is a global variable provided in
+  # Update-DocsMsPackages.ps1. Its value can set a "customSource" property.
+  # If it is empty then the property is not overridden.
+  $customPackageSource = Get-Variable -Name 'PackageSourceOverride' -ValueOnly -ErrorAction 'Ignore'
+  if (!$customPackageSource) {
+    return $package
+  }
+
+  if (!(Get-PackageSource -Name CustomPackageSource -ErrorAction SilentlyContinue)) {
+    Write-Host "Registering custom package source $customPackageSource"
+    Register-PackageSource `
+      -Name CustomPackageSource `
+      -Location $customPackageSource `
+      -ProviderName NuGet `
+      -Force
+  }
+
+  Write-Host "Checking custom package source for $($package.Name)"
+  $existingVersions = Find-Package `
+    -Name $package.Name `
+    -Source CustomPackageSource `
+    -AllVersions `
+    -AllowPrereleaseVersions
+  
+  # Matches package version against output: 
+  # "Azure.Security.KeyVault.Secrets 4.3.0-alpha.20210915.3"
+  $matchedVersion = $existingVersions.Where({$_.Version -eq $package.Versions})
+
+  if (!$matchedVersion) { 
```

Comment:
If we don't find a match should we give some error? Or perhaps check nuget.org and give an error if we don't find it there as well because the docs process is likely going to fail on this package.

Reply:
`EnsureCustomSource` is meant to validate that a package at a given version is in the custom feed WHEN a custom feed is specified. We didn't have to do this with JS. This means our list of validations is growing and a different pattern would probably be more readable here. 

To handle this in .NET I've got: https://github.com/Azure/azure-sdk-for-net/issues/24216

We'll probably encounter this again in Python and Java where we have to do an additional layer of validation. @sima-zhu